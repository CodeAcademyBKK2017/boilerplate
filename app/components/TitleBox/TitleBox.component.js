import React, {Component} from 'react';
import styles from './TitleBox.style';
import {
  Text,
  TextInput,
  View
} from 'react-native';

export default class TextArea extends Component {
  render () {
    return (
      <View style={styles.titleStyle}>
        <View style={styles.TitleStyleColum}>
          <Text style={styles.textTitle}>Note Title</Text>
          <View style={styles.languageTitle}><Text style={styles.textLanguage}>en</Text></View>
        </View>
        <TextInput 
          style={styles.textInput}
          placeholder={'Tasks for today'}
          placeholderTextColor={'#000'}
          underlineColorAndroid='transparent'
        />
      </View>
    );
  }
}