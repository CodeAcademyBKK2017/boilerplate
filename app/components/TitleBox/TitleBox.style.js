import {StyleSheet} from 'react-native';

export default StyleSheet.create({
  titleStyle: {
    width: '100%',
    paddingLeft: 10,
    paddingRight: 10
  },
  TitleStyleColum: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    paddingTop: 15,
    paddingBottom: 15
  },
  instructions: {
    color: '#333333'
  },
  textInput: {
    justifyContent: 'flex-start',
    alignItems: 'center',
    width: '100%',
    borderColor: 'gray', 
    borderWidth: 1,
    borderRadius: 3,
    backgroundColor: 'white',
    padding: 10,
    fontWeight: 'bold'
  },
  textTitle: {
    flex: 1,
    justifyContent: 'flex-start',
    fontWeight: 'bold'
  },
  languageTitle: {
    borderColor: 'gray', 
    borderRadius: 5,
    borderWidth: 1,
    backgroundColor: '#EEEEEE',
    padding: 8
  },
  textLanguage: {
    fontWeight: 'bold'
  }
});
