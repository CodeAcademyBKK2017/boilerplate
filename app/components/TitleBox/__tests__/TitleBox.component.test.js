import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import TitleBox from '../TitleBox.component';

describe('TitleBox', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <TitleBox />
    );
    expect(tree).toBeDefined();
  });

  it('TitleBox: Should return TextArea', () => { // example to test class methods
    const tree = renderer.create(<TitleBox />);
    expect(tree).toMatchSnapshot();
  });
});