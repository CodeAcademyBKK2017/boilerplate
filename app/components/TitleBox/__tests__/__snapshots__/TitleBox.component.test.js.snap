// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleBox TitleBox: Should return TextArea 1`] = `
<View
  style={
    Object {
      "paddingLeft": 10,
      "paddingRight": 10,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "paddingBottom": 15,
        "paddingTop": 15,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "flex": 1,
          "fontWeight": "bold",
          "justifyContent": "flex-start",
        }
      }
    >
      Note Title
    </Text>
    <View
      style={
        Object {
          "backgroundColor": "#EEEEEE",
          "borderColor": "gray",
          "borderRadius": 5,
          "borderWidth": 1,
          "padding": 8,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
        en
      </Text>
    </View>
  </View>
  <TextInput
    allowFontScaling={true}
    placeholder="Tasks for today"
    placeholderTextColor="#000"
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderColor": "gray",
        "borderRadius": 3,
        "borderWidth": 1,
        "fontWeight": "bold",
        "justifyContent": "flex-start",
        "padding": 10,
        "width": "100%",
      }
    }
    underlineColorAndroid="transparent"
  />
</View>
`;
